src/profile/external-identities/ExternalIdentities.test.tsx

```diff
@@ -52,4 +52,25 @@ describe('ExternalIdentities', () => {
       expect(screen.queryByText('GitHub')).toBeNull();
     });
   });
+  it('sorts providers based on desiredOrder', async () => {
+    // Arrange
+    const desiredOrder = ['ras', 'era-commons', 'fence', 'dcf-fence', 'kids-first', 'anvil', 'github'];
+    asMockedFn(getConfig).mockReturnValue({ externalCreds: { providers: desiredOrder } });
+
+    // Act
+    render(<ExternalIdentities queryParams={{}} />);
+
+    // Assert
+    const mainElement = screen.getByRole('main');
+    const providerElements = Array.from(mainElement.querySelectorAll('div')).map((div) => div.textContent);
+
+    expect(providerElements).toHaveLength(desiredOrder.length);
+    expect(providerElements[0]).toBe('NIH Researcher Auth Service (RAS)');
+    expect(providerElements[1]).toBe('NHLBI BioData Catalyst Framework Services');
+    expect(providerElements[2]).toBe('NCI CRDC Framework Services');
+    expect(providerElements[3]).toBe('Kids First DRC Framework Services');
+    expect(providerElements[4]).toBe('NHGRI AnVIL Data Commons Framework Services');
+    expect(providerElements[5]).toBe('eRA Commons');
+    expect(providerElements[6]).toBe('Nih Account');
```

Comment:
optional: it might be easier to debug test failures here if you considered the whole array in one expectation, something like:
```ts
    const expected = [
      'NIH Researcher Auth Service (RAS)',
      'NHLBI BioData Catalyst Framework Services',
      'NCI CRDC Framework Services',
      'Kids First DRC Framework Services',
      'NHGRI AnVIL Data Commons Framework Services',
      'eRA Commons',
      'Nih Account',
    ];

    expect(providerElements).toBe(expected);
```
your call; either way is fine

Reply:
Fixed: https://github.com/DataBiosphere/terra-ui/pull/5259/commits/5c98023b61bea1275c7142349319550142324d78